perceval/backends/core/gitter.py

```diff
@@ -0,0 +1,378 @@
+
+        logger.debug("Fetch process completed: %s messages fetched", num_msgs)
+
+    @classmethod
+    def has_archiving(cls):
+        """Returns whether it supports archiving items on the fetch process.
+
+        :returns: this backend supports items archive
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend does not support items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Gitter item."""
+
+        return item['id']
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts and coverts the sent time of a message
+        from a Gitter item.
+
+        The timestamp is extracted from 'sent' field and
+        converted to a UNIX timestamp.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = str_to_datetime(item['sent'])
+
+        return ts.timestamp()
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Gitter item.
+
+        This backend only generates one type of item which is
+        'message'.
+        """
+        return CATEGORY_MESSAGE
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return GitterClient(self.api_token, self.max_items, self.archive,
+                            self.sleep_for_rate, self.min_rate_to_sleep,
+                            from_archive, self.ssl_verify)
+
+
+class GitterClient(HttpClient, RateLimitHandler):
+    """Gitter API client.
+
+    Client for fetching information from the Gitter server
+    using its REST API.
+
+    :param api_token: key needed to use the API
+    :param max_items: maximum number of items per request
+    :param archive: an archive to store/read fetched data
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    """
+    AUTHORIZATION_HEADER = 'Authorization'
+    RMESSAGES = 'chatMessages'
+    RROOMS = 'rooms'
+
+    def __init__(self, api_token, max_items=MAX_ITEMS, archive=None,
+                 sleep_for_rate=False, min_rate_to_sleep=MIN_RATE_LIMIT,
+                 from_archive=False, ssl_verify=True):
+
+        base_url = GITTER_API_URL
+        self.api_token = api_token
+        self.max_items = max_items
+
+        super().__init__(base_url, archive=archive, from_archive=from_archive,
+                         ssl_verify=ssl_verify)
+        super().setup_rate_limit_handler(sleep_for_rate=sleep_for_rate, min_rate_to_sleep=min_rate_to_sleep)
+
+    def calculate_time_to_reset(self):
+        """Number of seconds to wait. They are contained in the rate limit reset header"""
+
+        time_to_reset = self.rate_limit_reset_ts - (datetime_utcnow().replace(microsecond=0).timestamp() + 1)
```

Comment:
When the rate limit is over, the time to reset is not logged correctly. Please check if the following code improves the log message:
```
-        time_to_reset = self.rate_limit_reset_ts - (datetime_utcnow().replace(microsecond=0).timestamp() + 1)
+        current_epoch = (datetime_utcnow().replace(microsecond=0).timestamp() + 1) * 1000
+        time_to_reset = (self.rate_limit_reset_ts - current_epoch) / 1000
```